addons/mail/models/mail_activity_mixin.py

```diff
@@ -281,7 +281,11 @@ def _read_group_groupby(self, groupby_spec, query):
             today_utc=pytz.utc.localize(datetime.utcnow()),
             tz=tz,
         )
-        alias = query.left_join(self._table, "id", sql_join, "res_id", "last_activity_state")
+        join_key = f"{self._table}__last_activity_state"
+        if join_key not in query._joins:
+            alias = query.left_join(self._table, "id", sql_join, "res_id", "last_activity_state")
+        else:
+            alias = join_key
```

Comment:
So in that case none of the work above is necessary right? Could we check that earlier?

I guess the issue is that it goes through this method twice for some reason, do we know why? Would it be possible to replicate in the tests? (see `test_mail_activity_mixin` in the `test_mail` module)